### DISCUSSION TITLE
Disappearing contents of page after user consents

### DISCUSSION THREAD

**Discussion by mkrawczykowski** [orestbida/cookieconsent#446] [Q&A] [answered]
Hi!
I have cookieconsent installed on my page with force_consent enabled. It generally works great, but I noticed a problem:
When user clicks on "Accept all" button, all the contents of the page disappear. The page is blank.
At the same time, when user chooses to consent to only some scripts (using the modal window with settings) and click Save button, everything works as expected.

However, I've found a probable reason for this: there is a script on my site that looks like this:

```
<script type="text/javascript" data-cookiecategory="analytics">
var kwDkUrl = 'https://c2s.pocketads.pl/d/[HERE IS MY ID]/default/?rnd=' + new String(Math.random()).substring(3, 9);document.write('<img style="border:none;width:1px;height:1px" src="'+kwDkUrl+'" referrerpolicy="no-referrer-when-downgrade" alt="" /'+'>');   
</script>
```

(notice the `document.write` part)

And when the page contents disappear, the HTML looks like this:

```
<html data-lt-installed="true">
  <head>
  <script style="display: none;">var tvt = tvt || {}; tvt.captureVariables = function(a){for(var b=
  new Date,c={},d=Object.keys(a||{}),e=0,f;f=d[e];e++)if(a.hasOwnProperty(f)&&"undefined"!=typeof a[f])try{var g=[];c[f]=JSON.stringify(a[f],function(h,k){try{if("function"!==typeof k){if("object"===typeof k&&null!==k){if(k instanceof HTMLElement||k instanceof Node||-1!=g.indexOf(k))return;g.push(k)}return k}}catch(m){}})}catch(h){}a=document.createEvent("CustomEvent");a.initCustomEvent("TvtRetrievedVariablesEvent",!0,!0,{variables:c,date:b});window.dispatchEvent(a)};window.setTimeout(function() {tvt.captureVariables({'dataLayer.hide': (function(a){a=a.split(".");for(var b=window,c=0;c<a.length&&(b=b[a[c]],b);c++);return b})('dataLayer.hide'),'gaData': window['gaData'],'dataLayer': window['dataLayer']})}, 2000);</script>
  </head>
  <body>
  <img style="border:none;width:1px;height:1px" src="https://c2s.pocketads.pl/d/[HERE IS MY ID]/default/?rnd=604115" referrerpolicy="no-referrer-when-downgrade" alt="">
  </body>
</html>
```

So I assume there is some kind of conflict between `document.write` from the above script and your cookieconsent, but I don't know how to solve it yet.

Could you please help?

**Comment by orestbida** [comment, marked as answer]
Hi @mkrawczykowski,

you can't use `document.write` once the page has finished loading; in other words, scripts calling `document.write` must not be deferred (which is what cookieconsent does).

If that is the only script using `document.write`, I'd highly suggest you use `document.body.insertAdjacentHTML` instead. This should do the exact same thing:
```html
<script type="text/plain" data-cookiecategory="analytics">
    var kwDkUrl = 'https://c2s.pocketads.pl/d/[HERE IS MY ID]/default/?rnd=' + new String(Math.random()).substring(3, 9);
    document.body.insertAdjacentHTML('beforeend', '<img style="border:none;width:1px;height:1px" src="' + kwDkUrl + '" referrerpolicy="no-referrer-when-downgrade" alt="" />');
</script>
```

If you have other scripts that depend on `document.write` you might want to load [postscribe](https://github.com/krux/postscribe), which will "tweak" `document.write` so that it works even if it is deferred.

**Comment by mkrawczykowski** [comment]
Wow, Orest, you're fast! :) Your solution really works! Thank you.
